.github/workflows/build-accumulo.yml

```diff
@@ -0,0 +1,96 @@
+name: Build Accumulo snapshot and update DataWave to use
+
+on:
+  workflow_dispatch:
+    inputs:
+      accumuloBranch:
+        required: false
+        description: "Branch name to build. Will be used as image tag."
+      accumuloRepo:
+        required: false
+        description: "Accumulo Repo to use. Expected to be at Github. Example: apache/accumulo"
+      deployAccumulo:
+        required: false
+        description: "Set to false if this accumulo version has already been pushed to Github Packages"
+
+# Defines two custom environment variables for the workflow. These are used for the Container registry domain, and a name for the Docker image that this workflow builds.
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository.lowercase }}
+  ACCUMULO_JAVA_VERSION: '17'
+  DATAWAVE_JAVA_VERSION: '11'
+  JAVA_DISTRIBUTION: 'zulu' #This is the default on v1 of the action for 1.8
+  USER_NAME: ${{ secrets.GHCR_WRITE_USER_NAME }}
+  ACCESS_TOKEN: ${{ secrets.GHCR_WRITE_ACCESS_TOKEN }}
+
+jobs:
+  build-and-deploy-accumulo:
+    runs-on: ubuntu-latest
+    # Sets the permissions granted to the `GITHUB_TOKEN` for the actions in this job.
+    permissions:
+      contents: read
+      packages: write
+      #
+    steps:
+    - name: Checkout DataWave
+      uses: actions/checkout@v4
+      with:
+        path: datawave
+
+    - name: Checkout Accumulo
+      uses: actions/checkout@v4.2.0
+      id: accumuloCheckout
+      with:
+        repository: ${{ github.event.inputs.accumuloRepo || 'apache/accumulo' }}
+        path: accumulo
+        ref: ${{ github.event.inputs.accumuloBranch || '2.1' }}
+
+    - name: Set up JDK ${{env.ACCUMULO_JAVA_VERSION}}
+      uses: actions/setup-java@v3
+      with:
+        distribution: ${{env.JAVA_DISTRIBUTION}}
+        java-version: ${{env.ACCUMULO_JAVA_VERSION}}
+        cache: 'maven'
+    - run: echo "ACCUMULO_JAVA=$JAVA_HOME" >> $GITHUB_ENV
+    - name: Set up JDK ${{env.DATAWAVE_JAVA_VERSION}}
+      uses: actions/setup-java@v3
+      with:
+        distribution: ${{env.JAVA_DISTRIBUTION}}
+        java-version: ${{env.DATAWAVE_JAVA_VERSION}}
+        cache: 'maven'
+    - run: echo "DATAWAVE_JAVA=$JAVA_HOME" >> $GITHUB_ENV
+
+    - name: Get Accumulo Version
+      id: get-accumulo-version
+      run: |
+        export JAVA_HOME=$ACCUMULO_JAVA
+        cd $GITHUB_WORKSPACE/accumulo
+        mvn build-helper:parse-version versions:set -DgenerateBackupPoms=false -DnewVersion=\${parsedVersion.majorVersion}.\${parsedVersion.minorVersion}.\${parsedVersion.incrementalVersion}-$(git rev-parse --short HEAD)
```

Comment:
So, this will dirty the workspace, which is fine, but the mavanagaiata plugin that Accumulo uses will put the sha1 in the MANIFEST.MF files in the jars, and if the workspace is dirty, it will append "-dirty" to it. Since you can't actually tell from that what changes in the workspace actually made it dirty, it'd be good to add some workflow identifier other than the sha1 to the name, to indicate that the version is associated with this particular workflow. And, from that, one could infer that the workspace is dirty in only the way associated with that particular workflow (in this case, only the version modification). For example, instead of `2.1.4-<commitId>`, you could do something like `2.1.4-nsa-<commitId>` or `2.1.4-datawave-<commitId>`, and then can infer that the build came from this particular workflow, and not another workflow that did something similar to put the commit ID in the version.

Reply:
I made it `-dwv-<commitId>` to signify it was built from this datawave repo. I didn't want to make it much longer just to ease length of the version number.